pkg/handler/handler.go

```diff
@@ -118,9 +120,22 @@ func addEnvToContainer(container *corev1.Container, mountPath, tokenName, volNam
 		Name:  "AWS_ROLE_ARN",
 		Value: roleName,
 	})
+
+	hostTokenFilePath := filepath.Join(mountPath, tokenName)
+	if isWindowsContainer {
+		// Convert the unix file path to a windows file path
+		// Eg. /var/run/secrets/eks.amazonaws.com/serviceaccount/token to
+		//     C:\var\run\secrets\eks.amazonaws.com\serviceaccount\token
+		if absolutePath, err := filepath.Abs(hostTokenFilePath); err != nil {
+			hostTokenFilePath = absolutePath
+		}
```

Comment:
If `filepath.Abs` returns an error, would the rest even work? Or would it try to refer to `"C:\tokenName"`?

Reply:
I'm not actually very sure as to what cases `filepath.Abs` would return the error. But perhaps we could handle it this way ...


```go
if absolutePath, err := filepath.Abs(hostTokenFilePath); err != nil {
	hostTokenFilePath = absolutePath
} else {
	if !strings.HasPrefix(hostTokenFilePath, "/") {
		hostTokenFilePath = "/" + hostTokenFilePath
	}
}
```

That way we ensure there is a leading slash even if there is an error. But no way to ensure the path is actually correct.

Any suggestions/insights from you on this ?

Reply:
Well, I think it would probably be better to just error out completely if that fails, since the token file path must be completely wrong at that stage.

Reply:
Yes. That seems to be a better option. Something like should work

```go
absolutePath, err := filepath.Abs(hostTokenFilePath)
if err != nil {
	klog.Errorf("error resolving absolute path of host token file: %v", err)
	return
}
hostTokenFilePath = "C:" + strings.Replace(absolutePath, `/`, `\`, -1)
```

Reply:
I was thinking something even more drastic, like `os.Exit(1)` instead of just a `return`. The token file path should really be correct.